src/domain/entities/figma-team.ts

```diff
@@ -1,20 +1,35 @@
+import type { ConnectUserInfo } from './connect-user-info';
+
 export enum FigmaTeamAuthStatus {
 	OK = 'OK',
 	ERROR = 'ERROR',
 }
 
-export type FigmaTeam = {
-	readonly id: string;
-	readonly webhookId: string;
-	readonly webhookPasscode: string;
-	readonly teamId: string;
-	readonly teamName: string;
-	readonly figmaAdminAtlassianUserId: string;
-	readonly authStatus: FigmaTeamAuthStatus;
-	readonly connectInstallationId: string;
-};
+export class FigmaTeam {
+	private readonly _adminInfo: ConnectUserInfo;
+
+	constructor(
```

Comment:
/nit given that there's a lot of string parameters in this constructor, it might be better to accept an object instead to avoid accidentally passing params in the wrong order:

```
export class FigmaTeam {
	private readonly _adminInfo: ConnectUserInfo;
	public readonly id: string;
  ...

  	constructor(params: {
		id: string;
    ...
	}) {
		this._adminInfo = {
			atlassianUserId: params.figmaAdminAtlassianUserId,
			connectInstallationId: params.connectInstallationId,
		};
		this.id = params.id;
    ...
	}
```

it is a bit more verbose though, so I'm happy to leave this as is if you don't think it's worth it

Reply:
If you did make this change, you could also update `FigmaTeamCreateParams` so it doesn't need to be updated if we ever add more getter properties:

```typescript
export type FigmaTeamCreateParams = Omit<
	ConstructorParameters<typeof FigmaTeam>[0],
	'id'
>;
```

Reply:
That's a very good point, Justin! I totally agree that it gets very easy to introduce a bug by accidentally swapping parameters. I've update the `FigmaTeam` constructor to accepts `params`.

With regards to `FigmaTeamCreateParams`, frankly speaking, I'm leaning towards explicitly listing the required fields for the following reasons:

- It is the simplest and easy to maintain solution.
- As you pointed out, `Omit<FigmaTeam, 'id'|'adminInfo'>;` requires updating the type when a new property is added to `FigmaTeam`.
- `Omit<ConstructorParameters<typeof FigmaTeam>[0], 'id'>` is a quite advance technique that can make the code more difficult to understand. Moreover, it can be subject to change as well, since there is no _direct_ relationship between `FigmaTeam` constructor parameters and `FigmaTeamCreateParams`. They are the same now but they don't have to be the same.

Please, let me know if you have any concerns, Justin.

Reply:
👍 makes sense!